rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -1051,7 +1051,18 @@ public static String toString(Object val) {
     static String defaultObjectToString(Scriptable obj) {
         if (obj == null) return "[object Null]";
         if (Undefined.isUndefined(obj)) return "[object Undefined]";
-        return "[object " + obj.getClassName() + ']';
+
+        String tag = null;
+        Object tagValue = ScriptableObject.getProperty(obj, SymbolKey.TO_STRING_TAG);
+        if (tagValue != Scriptable.NOT_FOUND && tagValue instanceof CharSequence) {
+            tag = tagValue.toString();
```

Comment:
what about doing a return here

    return "[object " + tagValue.toString() + "]";

Then you do not need to have the tag var at all and some more if's are gone for the price to have two static strings twice (but the compiler will take care of that).

Reply:
Thanks for your feedback! As you advised, now returning earlier.